File: block.py
import color

class Block:
    def __init__(self, color):
        self.color = color


DANDELION = Block(color.PLANT)
ROSE = Block(color.PLANT)
MUSHROOM_RED = Block(color.RED)
MUSHROOM_BROWN = Block(color.BROWN)
AIR = Block(color.NONE)
LEAVES = Block(color.PLANT)
TRUNK = Block(color.WOOD)
LAVA = Block(color.FIRE)
STONE = Block(color.STONE)
DIRT = Block(color.DIRT)
WATER = Block(color.WATER)
GRAVEL = Block(color.STONE)
SAND = Block(color.SAND)
GRASS = Block(color.GRASS)

File: Java/Block.java
public class Block {
  Color color;
  Ore ore;
  double abundance;
  
  public Block(Color color) {
    this.color = color;
  }

  public Block(Ore ore) {
    this.ore = ore;
    this.color = ore.color;
    this.abundance = ore.abundance;
  }
}
